spec/requests/foods_spec.rb

```diff
@@ -0,0 +1,129 @@
+require 'rails_helper'
+
+# This spec was generated by rspec-rails when you ran the scaffold generator.
+# It demonstrates how one might use RSpec to test the controller code that
+# was generated by Rails when you ran the scaffold generator.
+#
+# It assumes that the implementation code is generated by the rails scaffold
+# generator. If you are using any extension libraries to generate different
+# controller code, this generated spec may or may not pass.
+#
+# It only uses APIs available in rails and/or rspec-rails. There are a number
+# of tools you can use to make these specs even more expressive, but we're
+# sticking to rails and rspec-rails APIs to keep things simple and stable.
+
+RSpec.describe '/foods', type: :request do
+  # Food. As you add validations to Food, be sure to
+  # adjust the attributes here as well.
+  let(:valid_attributes) do
+    skip('Add a hash of attributes valid for your model')
+  end
+
+  let(:invalid_attributes) do
+    skip('Add a hash of attributes invalid for your model')
+  end
+
+  describe 'GET /index' do
+    it 'renders a successful response' do
+      Food.create! valid_attributes
+      get foods_url
+      expect(response).to be_successful
+    end
+  end
```

Comment:
- [ ] All the requests tests are failing because of one thing, the path they redirect to is the sign_in path. So all expectations will fail because there is no user sign_in step include in the tests. If you do 👉🏼 **_`puts(response.body)`_**, you will get this 👇🏼:
       ![redirect](https://user-images.githubusercontent.com/45002654/141779415-e665687f-544c-467b-bce6-8213efd627e5.png)
       - You should create a user, log in the user, create food using **_user.foods_**, then now expect that we will get redirected foods_path. For example:

```suggestion
  describe 'GET /index' do
    it 'renders a successful response' do
      user = User.create! valid_attributes_for_user
      sign_in user # capybara will gladly do this step for you
      user.foods.create! valid_attributes
      get foods_url
      expect(response).to be_successful
    end
  end
```

You may use this example to refactor all request tests. I took time and refactored your code and everything was fine. Just ensure that capybara is added to your Gemfile and installed through b**_`undle install`_**.